### PR TITLE
Fix global setting reference for max secondary storage

### PR DESCRIPTION
**Pull request by harikrishna-patnala**
This PR fixes issue #6411 

There are two global settings max.account.secondary.storage and max.project.secondary.storage to specify maximum secondary storage space (in GiB) that can be used for an account and project respectively.

But during upload template from project is not considering max.project.secondary.storage but uses max.account.secondary.storage.

We have fixed this here to use max.project.secondary.storage in case of project.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### How Has This Been Tested?

1. Uploaded a template from both account and project and observed using appropriate global settings.

### CODE REVIEW

File: core/src/main/java/org/apache/cloudstack/storage/command/TemplateOrVolumePostUploadCommand.java [hunk 1]
```diff
@@ -49,7 +49,7 @@ public class TemplateOrVolumePostUploadCommand {
 
     String description;
 
-    private String defaultMaxAccountSecondaryStorage;
+    private String defaultMaxSecondaryStorage;
```
Comment (sureshanaparti) [inline core/src/main/java/org/apache/cloudstack/storage/command/TemplateOrVolumePostUploadCommand.java:52]:
```suggestion
    private String defaultMaxSecondaryStorageInGB;
```

update relevant methods as well, rest LGTM.

**Review by sureshanaparti — APPROVED** [review]

**Comment by sureshanaparti** [comment]
I think, it's better to merge this in 4.16 branch as well (if so, change target to 4.16 and forward merge to 4.17 / main, your thoughts pls @nvazquez @harikrishna-patnala).

File: core/src/main/java/org/apache/cloudstack/storage/command/TemplateOrVolumePostUploadCommand.java (hunk 1 above)
Comment (harikrishna-patnala) [reply core/src/main/java/org/apache/cloudstack/storage/command/TemplateOrVolumePostUploadCommand.java]: changed the naming

**Comment by DaanHoogland** [comment]
@harikrishna-patnala is it a big issue to change these to `ConfigKey`s? the `Config` calles are deprected and we do need to remove this old code sometime (sooner is better imho)

File: api/src/main/java/com/cloud/user/ResourceLimitService.java [hunk 2]
```diff
@@ -27,6 +27,10 @@
 
 public interface ResourceLimitService {
 
+    static final ConfigKey<Long> MaxAccountSecondaryStorage = new ConfigKey<Long>("Account Defaults", Long.class, "max.account.secondary.storage", "400",
+            "The default maximum secondary storage space (in GiB) that can be used for an account", false);
+    static final ConfigKey<Long> MaxProjectSecondaryStorage = new ConfigKey<Long>("Project Defaults", Long.class, "max.project.secondary.storage", "400",
+            "The default maximum secondary storage space (in GiB) that can be used for an project", false);
```
Comment (DaanHoogland) [inline api/src/main/java/com/cloud/user/ResourceLimitService.java:33]:
```suggestion
            "The default maximum secondary storage space (in GiB) that can be used for a project", false);
```

**Comment by harikrishna-patnala** [comment]
> @harikrishna-patnala is it a big issue to change these to `ConfigKey`s? the `Config` calles are deprected and we do need to remove this old code sometime (sooner is better imho)

Replaced with configkeys, please review @DaanHoogland

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Comment by harikrishna-patnala** [comment]
> I think, it's better to merge this in 4.16 branch as well (if so, change target to 4.16 and forward merge to 4.17 / main, your thoughts pls @nvazquez @harikrishna-patnala).
It's not a critical or regularly hit issue so I think it's not that necessary to go into 4.16, we can treat this as a regular bug.

File: api/src/main/java/com/cloud/user/ResourceLimitService.java [hunk 3]
```diff
@@ -27,6 +27,10 @@
 
 public interface ResourceLimitService {
 
+    static final ConfigKey<Long> MaxAccountSecondaryStorage = new ConfigKey<Long>("Account Defaults", Long.class, "max.account.secondary.storage", "400",
```
[the remark this answers is not in the document]
Comment (harikrishna-patnala) [reply api/src/main/java/com/cloud/user/ResourceLimitService.java]: Fixed it.

**Comment by sureshanaparti** [comment]
@blueorangutan package

**Comment by sureshanaparti** [comment]
@blueorangutan package

**Comment by sureshanaparti** [comment]
@blueorangutan test

**Pull request merged**